libmachine/ssh/client.go

```diff
@@ -89,18 +90,18 @@ func SetDefaultClient(clientType ClientType) {
 	}
 }
 
-func NewClient(user string, host string, port int, auth *Auth) (Client, error) {
+func NewClient(user string, host string, port int, maxAttempt int, auth *Auth) (Client, error) {
```

Comment:
Not sure if there is an easy way around it, but I find it strange to have this setting at the ssh client level, feels more like external logic.